pkg/pex/option.go

```diff
@@ -82,13 +87,16 @@ func WithDiscovery(d discovery.Discovery, opt ...discovery.Option) Option {
 //
 // If d <= 0, a default value of 1m is used.  Users
 // SHOULD NOT alter this value without good reason.
-func WithTick(d time.Duration) Option {
-	if d <= 0 {
-		d = time.Minute
+func WithTick(tf func(ns string) time.Duration) Option {
```

Comment:
Same as above:  `d` -> `tf`.

I also suggest renaming `tf` to `f`, since it's not clear from the context that "f" stands for "Factory".

Reply:
The more I look at this, the more I winder if it belongs in `GossipParams`.  What do you think?